src/core/app.lua

```diff
@@ -51,10 +51,13 @@ local function with_restart (app, method)
 end
 
 -- Restart dead apps.
+local actions = { start={}, restart={}, reconfig={}, keep={}, stop={} }
 function restart_dead_apps ()
+   if not use_restart then return end
    local restart_delay = 2 -- seconds
-   local actions = { start={}, restart={}, reconfig={}, keep={}, stop={} }
    local restart = false
+   actions.restart[1] = nil
```

Comment:
> I guess the proper way to clear the array is to traverse it and set all elements to nil.

In plain Lua it's the only way, but LuaJIT (2.1) includes [`table.clear(t)`](http://repo.or.cz/w/luajit-2.0.git/blob_plain/v2.1:/doc/extensions.html) that does exactly that.  Of course, it has a lot of warnings that it's only for very specific cases, usually it's better to let the GC do its magic, etc.